src/passes/MemoryPacking.cpp

```diff
@@ -125,26 +125,36 @@ struct MemoryPacking : public Pass {
     module->memory.segments.swap(packed);
   }
 
-  void optimizeTrappingBulkMemoryOps(Module* module) {
-    struct Trapper : PostWalker<Trapper> {
-      bool changed = false;
-      void process(Expression* curr, Index index) {
-        if (!getModule()->memory.segments[index].isPassive) {
+  void optimizeTrappingBulkMemoryOps(PassRunner* runner, Module* module) {
+    struct Trapper : WalkerPass<PostWalker<Trapper>> {
+      bool changed;
+      void visitMemoryInit(MemoryInit* curr) {
+        if (!getModule()->memory.segments[curr->segment].isPassive) {
+          Builder builder(*getModule());
+          replaceCurrent(builder.blockify(builder.makeDrop(curr->dest),
+                                          builder.makeDrop(curr->offset),
+                                          builder.makeDrop(curr->size),
+                                          builder.makeUnreachable()));
+          changed = true;
+        }
+      }
+      void visitDataDrop(DataDrop* curr) {
+        if (!getModule()->memory.segments[curr->segment].isPassive) {
           ExpressionManipulator::unreachable(curr);
           changed = true;
         }
       }
-      void visitMemoryInit(MemoryInit* curr) { process(curr, curr->segment); }
-      void visitDataDrop(DataDrop* curr) { process(curr, curr->segment); }
       void walkFunction(Function* func) {
         changed = false;
         PostWalker<Trapper>::walkFunction(func);
         if (changed) {
           ReFinalize().walkFunctionInModule(func, getModule());
         }
       }
+      bool isFunctionParallel() override { return true; }
+      Pass* create() override { return new Trapper; }
     } trapper;
-    trapper.walkModule(module);
+    trapper.run(runner, module);
```

Comment:
this is confusing actually - this just does the normal walk. I should add an assertion about it. To parallelize you need to create a `PassRunner`, add the pass, and run that. Should also set `isNested`. See for example Inlining which has nested passes.

Reply:
Oh oops lol